### PR TITLE
Add `Mutation.contentUserRegistration`

### PR DESCRIPTION
**Pull request by B77Mills**
Add the new UserRegistration fields to the graph definition & resolvers also add the mutation support for UserRegistration

### CODE REVIEW

**Comment by B77Mills** [comment]
[Content UserRegistration by date management components](https://trello.com/c/IrcXjRpf/10-content-userregistration-by-date-management-components)

File: services/graphql-server/src/graphql/definitions/platform/content/index.js [hunk 1]
```diff
@@ -472,6 +478,28 @@ input ContentEventDatesMutationInput {
...
+  "Is Registration Required"
+  isRequired: Boolean
+  "Date to start require registration"
+  startDate: Date
+  "Date to end required registration"
+  endDate: Date
+  "List of site ids to enable required registration"
+  siteIds: [ObjectID!]
+  "Which level of access is required"
+  accessLevels: [String]
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/index.js]: Ensure all array items are non-nullable and all defaults are set to empty, so:

`accessLevels: [String!] = []`
`siteIds: [ObjectID!]! = []`

File: services/graphql-server/src/graphql/definitions/platform/content/index.js [hunk 2]
```diff
@@ -472,6 +478,28 @@ input ContentEventDatesMutationInput {
...
+  "Content Id"
+  id: Int!
+  "Payload to update"
+  payload: ContentUserRegistrationMutationPayloadInput!
+}
+
+"User Registration fields to update on content by id"
+input ContentUserRegistrationMutationPayloadInput {
+  "Is Registration Required"
+  isRequired: Boolean
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/index.js]: `isRequired` should be non-nullable

File: services/graphql-server/src/graphql/definitions/platform/content/index.js [hunk 3]
```diff
@@ -472,6 +478,28 @@ input ContentEventDatesMutationInput {
...
+input ContentUserRegistrationMutationInput {
+  "Content Id"
+  id: Int!
+  "Payload to update"
+  payload: ContentUserRegistrationMutationPayloadInput!
+}
+
+"User Registration fields to update on content by id"
+input ContentUserRegistrationMutationPayloadInput {
+  "Is Registration Required"
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/index.js]: Clean up the comments to be more declarative.. e.g. "Whether registration is required to view the content"

And for dates.... "The date when the registration requirement begins" etc

File: services/graphql-server/src/graphql/definitions/platform/content/index.js (hunk 1 above)
Comment (zarathustra323) [reply services/graphql-server/src/graphql/definitions/platform/content/index.js]: Actually, it doesn't look like the resolver is handling access levels or site IDs, so just remove these from the input definition for now

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**